File: packages/shared/src/types/source.ts
import type { SourceId, WorkspaceId } from "./ids.js";
import type { SourceKind, SourceStatus } from "./enums.js";

export interface SourceManifest {
  readonly ingestedAt: string;
  readonly normalizedAt: string | null;
  readonly originUrl: string | null;
  readonly mimeType: string | null;
  readonly fileSizeBytes: number | null;
  readonly chunkCount: number | null;
  readonly model: string | null;
  readonly confidenceNotes: string | null;
  readonly needsReingest?: boolean;
}

export interface Source {
  readonly id: SourceId;
  readonly workspaceId: WorkspaceId;
  readonly kind: SourceKind;
  readonly status: SourceStatus;
  readonly title: string;
  readonly description: string | null;
  readonly storageKey: string;
  readonly manifest: SourceManifest | null;
  readonly createdAt: string;
  readonly updatedAt: string;
}

File: services/api/src/atlas_api/models/source.py
"""Source domain models — mirrors packages/shared/src/types/source.ts."""

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import SourceKind, SourceStatus


class SourceManifest(BaseModel):
    model_config = ConfigDict(frozen=True)

    ingested_at: str
    normalized_at: str | None
    origin_url: str | None
    mime_type: str | None
    file_size_bytes: int | None
    chunk_count: int | None
    model: str | None
    confidence_notes: str | None
    needs_reingest: bool = False


class Source(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    kind: SourceKind
    status: SourceStatus
    title: str
    description: str | None
    storage_key: str
    manifest: SourceManifest | None
    created_at: str
    updated_at: str
